### ISSUE TITLE
Building OpenJFX webpage suggestions

### ISSUE THREAD

**Issue by Maxoudela** [javafxports/openjdk-jfx#156]
I've been following the Building OpenJFX web page (https://wiki.openjdk.java.net/display/OpenJFX/Building+OpenJFX#BuildingOpenJFX-custom) and I may have several suggestions to make.

First of all, thanks for all the information on that page, it really helped me. I consider myself as a total newbie so I apologize in advance if my suggestions are a bit naive. I'll let you be the judge for them. But if I stumbled on an issue, there's a chance someone else would. And this could be a chance to clarify some things to allow more people to contribute to OpenJFX.

I'm using Windows 10, in a random order:

- It is said to install

> Microsoft Visual Studio 2017, either Professional or Community edition

Well basically, I did that but I think we should explicitly say what is needed. From what I understand "C++ desktop develop" package must be checked and installed otherwise no "VC" folder will appear in the installation path.
When I ran the installer, I have plenty of choices and I wasn't sure which ones were necessary : https://imgur.com/a/0AMp4FX

Still on the subject, it is later said :

> You will likely need to set the following env variables to point to your VS 2017 installation, since Microsoft no longer sets such variables. This presumes that  you are using build 14.11.25503 of the Community edition of Visual Studio 2017. You should adjust it as needed for your system.

What I did was to open Visual Studio, clicked on "about" and copy my version. Spoiler alert : don't do that.
What I finally did was to go to "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC" and read the version of the folder in that location. Maybe there's a better way to do that?

- It should be explicitly said that all bash inputs should be done on Cygwin terminal. I tried to put 
`"$JAVA_HOME/bin/java" -version`
into a Windows CMD and it was not working. Same goes with the :
`export VS150COMNTOOLS="C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Auxiliary\\Build"`
As a side-note, I ended up installing these variable using the windows environment variable tool, which has been improved in Windows 10.

- There is something unclear about the root project. We first say

> The "root" project is "javafx".
Later it is said 

> I mentioned above that our root project is called "rt"
And later on, it is said 

> So the root project is simply named "rt" (or whatever your top directory is named).
In my case, the root project is "openjdk-jfx" I believe. That's a bit confusing but I'm really nit-picking here.

- Maybe update the list of projects? In my case, I do not have "fxpackager" nor "fxpackagerservices" nor "jmx". But maybe these projects are disabled by default and not appearing in my case? 

- It is said :

> Microsoft DirectShow header files – If you build media you will need the DirectShow header files from the Microsoft Windows SDK v7.1, installed in its default location of "C:\Program Files\Microsoft SDKs\Windows\v7.1"

I believed I have Microsoft SDK 10 installed on my computer. But I have nothing under "Program Files". I only have some "ClickOnce Bootstrapper", "Portable" , "Windows Kits" and "Windows Phone" folders under "C:\Program Files (x86)\Microsoft SDKs". So that's a bit unclear to me if I should specifically install Windows 7.1 SDK. (Maybe the reason my build is currently failing).

- In the Common Prerequisites, it is said to check for Gradle version and then Apache ant Version. I used Gradle 4.8 version because I'm building with a JDK 11 version and I was not able to use Gradle 4.6 because of this : https://github.com/gradle/gradle/issues/4515

But when you do :
`gradle -version`
Gradle is printing you a version of Ant. Apparently, it's an inner ant version that has nothing to do with my own ant version. If I do "ant -version" I do have the recommanded 1.8.2. But I must admit I've search a bit on how to get a gradle version that has the right version of ant..

I'm finished with all my notes because my build is failing and I haven't been able to go beyond. 
I let you judge if they are relevant. Anyhow, others may find my explication useful later.

**Comment by johanvos** [comment]
thank you for your notes. The build instructions are indeed a bit outdated, we have to improve these. Your suggestions are very useful, thanks for that.

**Comment by Maxoudela** [comment]
Since I switch to Mac in order to build OpenJFX, here is another comment : 

> Xcode developer command line tools – you can install them by using the menus within Xcode: XCode -> Preferences -> Downloads -> Components

I downloaded Xcode directly from the App store and I believe the command line tools was already shipped with it. I did not managed to find the Downloads menu: https://imgur.com/a/Ojtb1HX

**Comment by nlisker** [comment]
Yes, Iv'e already mentioned some of that last year on the mailing list: http://mail.openjdk.java.net/pipermail/openjfx-dev/2017-December/021197.html (keep reading till the happy ending). I think it was forgotten because of the holidays.

I think that not many people are going to bother with this now that this GitHub repo is up and running though.

**Comment by kevinrushforth** [comment]
> I think that not many people are going to bother with this now that this GitHub repo is up and running though.

Developers who use Windows will still need a local build of the native libs to build and test anything. So it is still useful / helpful to improve the docs (and, more importantly, to make the builds more forgiving / bullet-proof).

**Comment by nlisker** [comment]
Yes, you are right.

**Comment by brcolow** [comment]
I have contributed [this script](https://github.com/javafxports/openjdk-jfx/blob/develop/tools/scripts/build.ps1) to make building locally on Windows more painless. Improvements to it are very welcome (at least from me :))

**Comment by nlisker** [comment]
@Maxoudela I updated the build page. Should be much more accurate now.
